frontend/public/module/k8s/resource.js

```diff
@@ -186,3 +188,48 @@ export const k8sWatch = (kind, query = {}, wsOptions = {}) => {
   wsOptions.path = path;
   return new WSFactory(path, wsOptions);
 };
+
+/**
+ * Use k8sWatch to wait for a resource to get into an expected condition.
+ * Watches for resource by kind, namespace and optional name.
+ * Promise resolves to a new resource version or rejects with a timeout.
+ * @param {K8sKind} kind
+ * @param {K8sResourceKind} resource
+ * @param {(kind: K8sResourceKind) => boolean} checkCondition
+ * @param {number} timeoutInMs
+ * @returns {Promise<K8sResourceKind>}
+ */
+export const k8sWaitForUpdate = (kind, resource, checkCondition, timeoutInMs) => {
+  const { namespace, name, resourceVersion } = resource.metadata;
+
+  if (checkCondition(resource)) {
+    return Promise.resolve(resource);
+  }
+
+  const watcher = k8sWatch(kind, {
+    ns: namespace,
+    resourceVersion,
+  });
+  const closeConnection = () => watcher.destroy(0);
+
+  const waitForCondition = new Promise((resolve, reject) => {
+    watcher.onbulkmessage((messages) => {
+      messages.forEach(({ object }) => {
+        if ((!name || name === object.metadata?.name) && checkCondition(object)) {
+          resolve(object);
+        }
+      });
+    });
+    watcher.onclose(() => reject(new Error('Connection closed')));
+    watcher.ondestroy(() => reject(new Error('Connection destroyed')));
+  });
+
+  const waitForTimeout = new Promise((resolve, reject) => {
+    setTimeout(
+      () => reject(new Error('Waiting for resource condition change timeout')),
```

Comment:
```suggestion
      () => reject(new Error('Timed out waiting for resource to finish')),
```
Past tense as it's already happened.